pandas/tests/indexes/datetimes/test_date_range.py

```diff
@@ -162,6 +162,21 @@ def test_date_range_ambiguous_arguments(self):
         with tm.assert_raises_regex(ValueError, msg):
             date_range(start, end, periods=10, freq='s')
 
+    def test_date_range_convenience_periods(self):
```

Comment:
Could you also test this with the `tz` arg specified? Would also be good to test a `tz` where there is a day light savings transition between `start` and `end`.